.github/workflows/ci.yml

```diff
@@ -0,0 +1,101 @@
+name: CI
+
+on:
+  push:
+  pull_request:
+  # Allow manual runs. It is necessary to run the workflow manually, when the "formatter" workflow applies any changes.
+  workflow_dispatch:
+
+jobs:
+  Processor_Partial:
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.id != github.event.pull_request.base.repo.id
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            framework: net9.0
+          - os: ubuntu-latest
+            framework: net8.0
+          - os: windows-latest
+            framework: net9.0
+          - os: windows-latest
+            framework: net8.0
+          - os: windows-latest
+            framework: net462
+          - os: macos-latest
+            framework: net9.0
+          - os: macos-latest
+            framework: net8.0
+          - os: macos-13
+            framework: net9.0
+          - os: macos-13
+            framework: net8.0
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup .NET SDK v9.0.x
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: 9.0.x
+      - name: Setup .NET SDK v8.0.x (if needed)
+        if: ${{ matrix.framework == 'net8.0' }}
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: 8.0.x
+      - name: Build
+        run: dotnet build G-Research.OpenTelemetry.Processor.Partial --configuration=Release -v=Normal --framework=${{ matrix.framework }}
```

Comment:
I would remove the build step, as it will be automatically triggered by the tests anyway